### PR TITLE
Add util.ModuleExists

### PR DESCRIPTION
**Pull request by Down-s**
This would be a much easier way to quickly check if somebody has a binary module installed, instead of having to manually check the operating system/architecture for file.Exists.

### CODE REVIEW

File: garrysmod/lua/includes/extensions/util.lua [hunk 1]
```diff
@@ -94,6 +94,20 @@ function util.LocalToWorld( ent, lpos, bone )
 
 end
 
+--[[---------------------------------------------------------
+	Name: ModuleInstalled( name )
+	Desc: Checks if a binary module is installed
+-----------------------------------------------------------]]
+function util.ModuleInstalled( name )
+    
+    local realm = CLIENT and "cl" or "sv"
```
Comment (GitSparTV) [inline garrysmod/lua/includes/extensions/util.lua:103]: Nice, this also covers menu realm.

**Review by GitSparTV — APPROVED** [review]

**Comment by GitSparTV** [comment]
One thing, maybe for consistency it should be named as ModuleExists

**Comment by WilliamVenner** [comment]
Doesn't handle `gmsv_module_linux.dll`

**Comment by WilliamVenner** [comment]
Edge case: `gmsv_module_linux32.dll` works on x86-64, but not on main branch. Lol

(main branch has to be `gmsv_module_linux.dll`)

**Comment by GitSparTV** [comment]
Branch inconsistency?

**Comment by Down-s** [comment]
> Edge case: `gmsv_module_linux32.dll` works on x86-64, but not on main branch. Lol
> 
> (main branch has to be `gmsv_module_linux.dll`)

So if `BRANCH` is main on linux it should just ignore the arch?

**Comment by GitSparTV** [comment]
The thing is BRANCH is not reliable

**Comment by CornerPin** [comment]
This should also handle `gmsv_module_osx.dll`/`gmcl_module_osx.dll`.
https://wiki.facepunch.com/gmod/Creating_Binary_Modules#naminglocation

**Comment by WilliamVenner** [comment]
BRANCH doesn't even work on the server.

**Comment by Kefta** [comment]
I think this would be better as a global function since we don't have a module/lua library. Would be consistent with `require`. The name is also a bit ambiguous with `module` in the API but I think that's unavoidable.

**Comment by GitSparTV** [comment]
What if we just add second argument into require to not throw  error and instead return bool?

`require(str)`
`bool = require(str, true)` (nothrow version)

**Comment by swampservers** [comment]
seems like good idea

**Comment by garryspins** [comment]
that would be a far better solution to this

**Comment by Kefta** [comment]
Modifying more built-in functions is a step in the wrong direction.

**Comment by WilliamVenner** [comment]
still needs fixing to work for `gmsv_module_linux32.dll` on `x86-64` branch 32-bit :(

**Comment by Down-s** [comment]
Closing for #1896

**Pull request closed without merging**